### PR TITLE
API 24874 OAS update AllergyIntolerance

### PR DESCRIPTION
**Pull request by bcastillomartel-va**
[API-24874](https://vajira.max.gov/browse/API-24874)

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java [hunk 1]
```diff
@@ -14,7 +14,12 @@
 
 public interface Dstu2AllergyIntoleranceApi {
   @Operation(
-      summary = "Allergy Intolerance Read",
+      summary =
+          "Returns a specific allergy intolerance based on _id. (READ)"
+              + "\n●\tThis endpoint can return a known patient allergy related "
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java:19]: You can use a hyphen here, the swagger editor will handle converting it to a bullet:

`- This endpoint can return a known patient allergy related to medication, food, or other substances.\n`

File: data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java [hunk 2]
```diff
@@ -14,7 +14,12 @@
 
 public interface Dstu2AllergyIntoleranceApi {
   @Operation(
-      summary = "Allergy Intolerance Read",
+      summary =
+          "Returns a specific allergy intolerance based on _id. (READ)"
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java:18]: Wouldn't the `summary` be just this top line, with the rest going in `description`?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java [hunk 3]
```diff
@@ -14,7 +14,12 @@
...
+              + "\n●\tThis endpoint can return a known patient allergy related "
+              + "to medication, food, or other substances.\n●\tFollows the "
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java:20]: For readability, keep `\n` at the end of the line that it terminates

File: data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java (hunk 1 above)
Comment (bcastillomartel-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java:19]: `\n●\t` -> `-` right? @jblefkove-va

Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java:19]: You don't need the tab character, but you do still need the `\n`. 

Example: https://github.com/department-of-veterans-affairs/lighthouse-facilities/blob/master/facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v0/FacilitiesService.java#L23

File: data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java [hunk 4]
```diff
@@ -51,9 +54,15 @@ AllergyIntolerance allergyIntoleranceRead(
           String id);
 
   @Operation(
-      summary = "Allergy Intolerance Search",
+      summary =
+          "Searches for appointments using at least one required "
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java:58]: wrong branch?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java [hunk 5]
```diff
@@ -14,9 +14,12 @@
 
 public interface Dstu2AllergyIntoleranceApi {
   @Operation(
-      summary = "Allergy Intolerance Read",
+      summary = "Returns a specific allergy intolerance based on _id. (READ)",
       description =
-          "http://www.fhir.org/guides/argonaut/r2/StructureDefinition-argo-allergyintolerance.html",
+          "- This endpoint can return a known patient allergy related "
+              + "to medication, food, or other substances.\n"
+              + "- Follows the [US Core AllergyIntolerance Profile]"
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java:21]: Argonaut, not US Core

File: data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java [hunk 6]
```diff
@@ -83,22 +91,18 @@ AllergyIntolerance.Bundle allergyIntoleranceSearch(
...
+                      + "patient parameter is required. If _id is used, "
+                      + "no other parameters can be used.")
           String id,
-      @Parameter(
-              in = ParameterIn.QUERY,
-              name = "identifier",
-              description =
-                  "The logical identifier of the resource. Once assigned, this value "
-                      + "never changes.")
-          String identifier,
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/api/Dstu2AllergyIntoleranceApi.java:101]: For DSTU2 leave `identifier`... also you are not required to update these parameter descriptions

File: data-query/src/main/java/gov/va/api/health/dataquery/service/api/R4AllergyIntoleranceApi.java [hunk 7]
```diff
@@ -83,26 +92,25 @@ AllergyIntolerance.Bundle allergyIntoleranceSearch(
...
-                  "The logical identifier of the resource. Once assigned, this value "
-                      + "never changes.")
-          String identifier,
       @Parameter(
               in = ParameterIn.QUERY,
               name = "_lastUpdated",
-              description = "The date when allergy intolerance record was last updated.")
+              description =
+                  "The date when the record was last updated. Supported formats are: YYYY, "
+                      + " YYYY-MM,  YYYY-MM-DD,  YYYY-MM-DD'T'HH:MM:SSZ "
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/api/R4AllergyIntoleranceApi.java:112]: why the double-spacing?

Comment (bcastillomartel-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/api/R4AllergyIntoleranceApi.java:112]: This is from the spreadsheet. Should I keep those single spaced?

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**